Program.cs

```diff
@@ -1,3 +1,192 @@
+
+            if (int.TryParse(input, out quantity) && quantity >= 0)
+                break;
+
+            Console.WriteLine("Invalid quantity. Please enter a non-negative integer.");
+        }
+
+        bool success = inventory.AddProduct(name, price, quantity, out var error);
+        if (success)
+            Console.WriteLine("Product added successfully.");
+        else
+            Console.WriteLine($"Failed to add product. {error}");
+    }
+
+
+    public static void ViewAllProducts(Inventory inventory)
+    {
+        var items = inventory.GetProducts();
+        if (items.Count == 0)
+            Console.WriteLine("No products in inventory.");
+        else
+        {
+            Console.WriteLine("Name | Price | Quantity");
+            foreach (Product p in items)
+            {
+                Console.WriteLine($"{p.Name} | {p.Price} | {p.Quantity}");
+            }
+        }
+    }
+
+    private static void EditProduct(Inventory inventory)
+    {
+        var targetName = ReadNonEmpty("Enter the product name to edit: ");
+        if (!inventory.FindByName(targetName, out var product))
+        {
+            Console.WriteLine("Not found.");
+            return;
+        }
+
+        Console.WriteLine($"Editing '{product!.Name}' (Price: {product.Price}, Quantity: {product.Quantity})");
+        Console.WriteLine("Press Enter to keep the current value.");
+
+        Console.Write($"New name [{product.Name}]: ");
+        var nameInput = Console.ReadLine();
+        string? newName = string.IsNullOrWhiteSpace(nameInput) ? null : nameInput.Trim();
+
+        decimal? newPrice = null;
+        while (true)
+        {
+            Console.Write($"New price [{product.Price}]: ");
+            var s = Console.ReadLine();
+            if (string.IsNullOrWhiteSpace(s)) break;
+            if (decimal.TryParse(s, out var v) && v >= 0)
+            {
+                newPrice = v;
+                break;
+            }
+
+            Console.WriteLine("Invalid. Enter a non‑negative decimal, or press Enter to keep.");
+        }
+
+        int? newQty = null;
+        while (true)
+        {
+            Console.Write($"New quantity [{product.Quantity}]: ");
+            var s = Console.ReadLine();
+            if (string.IsNullOrWhiteSpace(s)) break;
+            if (int.TryParse(s, out var v) && v >= 0)
+            {
+                newQty = v;
+                break;
+            }
+
+            Console.WriteLine("Invalid. Enter a non‑negative integer, or press Enter to keep.");
+        }
+
+        bool success = inventory.EditProduct(product, newName, newPrice, newQty, out var error);
+        Console.WriteLine(success ? "Product edited successfully." : $"Failed to edit product. {error}");
+    }
+
+    private static void DeleteProduct(Inventory inventory)
+    {
+        var name = ReadNonEmpty("Enter the product name to delete: ");
+        bool success = inventory.DeleteProduct(name, out var error);
```

Comment:
i would prefer using meaningful names e.g.
```suggestion
        var isProductDeleted= inventory.DeleteProduct(name, out var error);
```